### ISSUE TITLE
Everything goes silent when I start pulseeffects

### ISSUE THREAD

**Issue by WarpspeedSCP** [wwmm/pulseeffects#148]
Here's the log- 

    ** (pulseeffects:3): WARNING **: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-OWhkFZfeqH: Connection refused
    20:03:40.846 - PulseEffects - INFO - pulseaudio context started
    20:03:40.852 - PulseEffects - INFO - connected to server: unix:/run/user/1000/pulse/native
    20:03:40.862 - PulseEffects - INFO - server protocol version: 32
    20:03:40.873 - PulseEffects - INFO - pulseaudio version: 10.0
    20:03:40.873 - PulseEffects - INFO - default pulseaudio source: alsa_input.pci-0000_00_1f.3.analog-stereo
    20:03:40.873 - PulseEffects - INFO - default pulseaudio sink: alsa_output.pci-0000_00_1f.3.analog-stereo
    20:03:40.873 - PulseEffects - INFO - default pulseaudio sink audio format: s16le
    20:03:40.873 - PulseEffects - INFO - default pulseaudio sink sampling rate: 48000 Hz. We will use the same rate.
    20:03:40.874 - PulseEffects - INFO - default pulseaudio source audio format: s16le
    20:03:40.874 - PulseEffects - INFO - default pulseaudio source sampling rate: 44100 Hz. We will use the same rate.
    20:03:40.874 - PulseEffects - INFO - loading Pulseeffects applications sink...
    20:03:40.878 - PulseEffects - INFO - sink owner module index: 25
    20:03:40.878 - PulseEffects - INFO - pulseaudio version: 10.0
    20:03:40.879 - PulseEffects - INFO - default pulseaudio source: alsa_input.pci-0000_00_1f.3.analog-stereo
    20:03:40.879 - PulseEffects - INFO - default pulseaudio sink: PulseEffects_apps
    20:03:40.879 - PulseEffects - INFO - Pulseeffects apps sink was successfully loaded
    20:03:40.879 - PulseEffects - INFO - Pulseeffects apps sink index:2
    20:03:40.879 - PulseEffects - INFO - Pulseeffects apps sink monitor name: PulseEffects_apps.monitor
    20:03:40.879 - PulseEffects - INFO - loading Pulseeffects microphone output sink...
    20:03:40.881 - PulseEffects - INFO - sink owner module index: 26
    20:03:40.881 - PulseEffects - INFO - Pulseeffects mic sink was successfully loaded
    20:03:40.881 - PulseEffects - INFO - pulseaudio version: 10.0
    20:03:40.881 - PulseEffects - INFO - default pulseaudio source: alsa_input.pci-0000_00_1f.3.analog-stereo
    20:03:40.881 - PulseEffects - INFO - default pulseaudio sink: PulseEffects_mic
    20:03:40.881 - PulseEffects - INFO - Pulseeffects mic sink index:3
    20:03:40.882 - PulseEffects - INFO - Pulseeffects mic sink monitor name: PulseEffects_mic.monitor
    20:03:40.899 - PulseEffects - INFO - limiter plugin was enabled
    20:03:40.899 - PulseEffects - INFO - compressor plugin was enabled
    20:03:40.899 - PulseEffects - INFO - highpass plugin was enabled
    20:03:40.900 - PulseEffects - INFO - lowpass plugin was enabled
    20:03:40.900 - PulseEffects - INFO - equalizer plugin was enabled
    20:03:40.900 - PulseEffects - INFO - reverb plugin was enabled
    20:03:40.900 - PulseEffects - INFO - spectrum plugin was enabled
    20:03:40.949 - PulseEffects - INFO - apps: exciter plugin was enabled
    20:03:40.949 - PulseEffects - INFO - apps: bass plugin was enabled
    20:03:40.950 - PulseEffects - INFO - apps: stereo plugin was enabled
    20:03:40.950 - PulseEffects - INFO - apps: stereo plugin was enabled
    20:03:40.950 - PulseEffects - INFO - apps: panorama plugin was enabled
    20:03:40.950 - PulseEffects - INFO - apps: maximizer plugin was enabled
    20:03:40.951 - PulseEffects - INFO - apps: output plugin was enabled
    20:03:40.956 - PulseEffects - INFO - limiter plugin was enabled
    20:03:40.957 - PulseEffects - INFO - compressor plugin was enabled
    20:03:40.957 - PulseEffects - INFO - highpass plugin was enabled
    20:03:40.957 - PulseEffects - INFO - lowpass plugin was enabled
    20:03:40.957 - PulseEffects - INFO - equalizer plugin was enabled
    20:03:40.957 - PulseEffects - INFO - reverb plugin was enabled
    20:03:40.958 - PulseEffects - INFO - spectrum plugin was enabled
    20:03:41.125 - PulseEffects - INFO - apps: bass plugin was enabled
    20:03:41.305 - PulseEffects - INFO - apps: spectrum plugin was enabled
    20:03:41.305 - PulseEffects - INFO - mic: spectrum plugin was enabled
    20:03:41.365 - PulseEffects - INFO - created stream for sink input 1
    20:03:41.477 - PulseEffects - INFO - pipeline state: playing
    20:03:41.566 - PulseEffects - INFO - pulsesrc latency-time [us]: 10000
    20:03:41.566 - PulseEffects - INFO - pulsesrc buffer-time [us]: 100000
    20:03:41.566 - PulseEffects - INFO - pulsesink latency-time [us]: 10000
    20:03:41.566 - PulseEffects - INFO - pulsesink buffer-time [us]: 100000
    20:04:03.623 - PulseEffects - INFO - apps: equalizer plugin was enabled
    20:04:04.848 - PulseEffects - INFO - apps: equalizer plugin was disabled
    20:04:32.541 - PulseEffects - INFO - pipeline state: null
    20:04:32.542 - PulseEffects - INFO - pipeline state: null
    20:04:32.542 - PulseEffects - INFO - sinks unloaded
    20:04:32.542 - PulseEffects - INFO - disconnecting pulseaudio context
    20:04:32.542 - PulseEffects - INFO - pulseaudio context terminated
    20:04:32.542 - PulseEffects - INFO - unferencing pulseaudio context object
    20:04:32.542 - PulseEffects - INFO - stream for sink input 1 was terminated
    20:04:32.542 - PulseEffects - INFO - stopping pulseaudio threaded main loop
    20:04:32.542 - PulseEffects - INFO - freeing pulseaudio main loop object

I tried toggling a thing or two but nothing happened...

The audio visualiser seemed to be showing that audio was playing, but the speakers didn't play anything.

Any idea about how to get around this?

**Comment by Ixoos** [comment]
If you are using Kde plasma
"I have no sound in KDE.
Answer: KDE needs Pulseaudio's module module-device-manager that conflicts with PulseEffects. See #99. This issue will be fixed in the next Pulseaudio's version (probably version 11.2) but until then you will have to unload this module before running PulseEffects pactl unload-module module-device-manager."

**Comment by WarpspeedSCP** [comment]
I'm on Gnome shell (pop_os).

**Comment by wwmm** [comment]
If you are not in kde the module `module-device-manager` will not be loaded. So the problem is somewhere else. In your case the problem is probably being caused by another module whose problems are also discussed in #99, module `module-switch-on-connect`. At startup PE asks Pulseaudio which is the default output device. In the log above we see that initially this device is `alsa_output.pci-0000_00_1f.3.analog-stereo`. But after sometime we see in the log that the default sink device was changed to `PulseEffects_apps`. This change was not done by PE but probably by the module `module-switch-on-connect`. Until the next version of Pulseaudio is released you will have to unload this module before starting PE:

`pactl unload-module module-switch-on-connect`

It seems that the latest ubuntu is not the only OS loading this modules by default. In case you want to make this change permanent you will have to edit the file `/etc/pulse/default.pa` and comment the line that loads this module

**Comment by WarpspeedSCP** [comment]
Yup, that's what did it. Thank you for the support!
